src/vs/workbench/contrib/chat/browser/chatListRenderer.ts

```diff
@@ -562,6 +557,10 @@ export class ChatAccessibilityProvider implements IListAccessibilityProvider<Cha
 	}
 }
 
+function countCodeBlocks(markdown: string): number {
+	return marked.lexer(markdown).filter(token => token.type === 'code')?.length ?? 0;
```

Comment:
You will need to walk the tree to find codeblocks that are inside of list items (rendering these actually has some issues). And there might be some other case where the codeblock is a child, not sure what that would be.

Reply:
Ok will see if that's raised in testing thanks